food_delivery/foodapp/views.py

```diff
@@ -1,3 +1,453 @@
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+class UserDetail(APIView):
+    def get_object(self,pk):
+        try:
+            return User.objects.get(pk=pk)
+        except User.DoesNotExist:
+            return None
+        
+    def get(self,request,pk):
+        user=self.get_object(pk)
+        if user is None:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+        serializer = UserSerializer(user)
+        return Response(serializer.data)
+
+    def put(self, request, pk):
+        user = self.get_object(pk)
+        if user is None:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+        serializer = UserSerializer(user, data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    def delete(self, request, pk):
+        user = self.get_object(pk)
+        if user is None:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+        user.delete()
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+class RestaurantListCreate(APIView):
+    def get(self, request):
+        restaurants = Restaurant.objects.all()
+        serializer = RestaurantSerializer(restaurants, many=True)
+        return Response(serializer.data)
+
+    def post(self, request):
+        serializer = RestaurantSerializer(data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+class RestaurantDetail(APIView):
+    def get_object(self, pk):
+        try:
+            return Restaurant.objects.get(pk=pk)
+        except Restaurant.DoesNotExist:
+            return None
+
+    def get(self, request, pk):
+        restaurant = self.get_object(pk)
+        if restaurant is None:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+        serializer = RestaurantSerializer(restaurant)
+        return Response(serializer.data)
+
+    def put(self, request, pk):
+        restaurant = self.get_object(pk)
+        if restaurant is None:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+        serializer = RestaurantSerializer(restaurant, data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    def delete(self, request, pk):
+        restaurant = self.get_object(pk)
+        if restaurant is None:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+        restaurant.delete()
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+class MenuItemListCreate(APIView):
+    def get(self, request):
+        menu_items = MenuItem.objects.all()
```

Comment:
you need to filter based on restaurant. should not show all the menu

Reply:
Done sir.